_posts/2017-11-21-release-1.4-and-1.5-timeline.md

```diff
@@ -0,0 +1,136 @@
+lots of credit for its hard work over the past few months, and we’re excited to see these features
+in the hands of users.
+
+This post will describe how the community plans to get there and the rationale behind the approach.
+
+## Coming soon: Major Changes to the Flink’s Runtime
+
+There are 3 significant improvements to the Apache Flink engine that the community has nearly
+completed and that will have a meaningful impact on Flink’s operability and performance.
+
+1. Rework of the deployment model and distributed process
+2. Transition to event-driven network I/O and application-level flow control for better backpressure handling
+3. Faster recovery from failure
+
+Next, we’ll go through each of these improvements in more detail.
+
+## Reworking Flink’s Deployment Model and Distributed Process
+
+[FLIP-6](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=65147077) is an initiative
+that’s been in the works for more than a year and represents a major refactor of Flink’s deployment
+model and distributed process. The underlying motivation for FLIP-6 was the fact that Flink is being
+adopted by a wider range of developer communities--both developers coming from the big data and
+analytics space as well as developers coming from the event-driven applications space.
+
+Modern, stateful stream processing has served as a convergence for these two developer communities.
+Despite a significant overlap of the core concepts in the applications being built, each group of
+developers has its own set of common tools, deployment models, and expected behavior when working
+with a stream processing framework like Flink.
+
+FLIP-6 will ensure that Flink fits naturally in both of these contexts, behaving as though it’s
+native to each ecosystem and operating seamlessly within a broader technology stack. A few of the
+specific changes in FLIP-6 that will have such an impact:
+
+ - Leveraging cluster management frameworks to support full resource elasticity
+ - First-class support for containerized environments such as Kubernetes and Docker
+ - REST-based client-cluster communication to ease operations and 3rd party integrations
+
+FLIP-6, along with already-introduced features like
+[rescalable state](https://data-artisans.com/blog/apache-flink-at-mediamath-rescaling-stateful-applications),
+lays the groundwork for dynamic scaling in Flink, meaning that Flink programs will be able to scale up or down
+automatically based on required resources--a huge step forward in terms of ease of operability and
+the efficiency of Flink applications.
+
+## Lower Latency via Improvements to the Apache Flink Network Stack
+
+Speed will always be a key consideration for users who build stream processing applications, and
+Flink 1.5 will include a rework of the network stack that will even further improve Flink's latency.
+At the heart of this work is a transition to event-driven network I/O, ensuring that Flink will use
```

Comment:
can you explain here this is a transition from what to event-driven network I/O?

Reply:
@aljoscha @bowenli86 proposed change to this section, similar to my comment above. 

"At the heart of this work is a transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control, ensuring that Flink..."